src/contract-state.ts

```diff
@@ -16,34 +16,75 @@
  */
 import { WalletAddress } from './common.js';
 
-export const AR_IO_CONTRACT_FUNCTIONS = {
-  GATEWAY: 'gateway',
-  GATEWAYS: 'gateways',
-  PRESCRIBED_OBSERVERS: 'prescribedObservers',
-  DELEGATE_STAKE: 'delegateStake',
-  DECREASE_DELEGATE_STAKE: 'decreaseDelegateStake',
-  JOIN_NETWORK: 'joinNetwork',
-  LEAVE_NETWORK: 'leaveNetwork',
-  INCREASE_OPERATOR_STAKE: 'increaseOperatorStake',
-  DECREASE_OPERATOR_STAKE: 'decreaseOperatorStake',
-  UPDATE_GATEWAY_SETTINGS: 'updateGatewaySettings',
-  SAVE_OBSERVATIONS: 'saveObservations',
-  SUBMIT_AUCTION_BID: 'submitAuctionBid',
-  BUY_RECORD: 'buyRecord',
-  EXTEND_RECORD: 'extendRecord',
-  INCREASE_UNDERNAME_COUNT: 'increaseUndernameCount',
-  RECORD: 'record',
-  AUCTION: 'auction',
-  TRANSFER: 'transfer',
-  VAULTED_TRANSFER: 'vaultedTransfer',
-  CREATE_VAULT: 'createVault',
-  EXTEND_VAULT: 'extendVault',
-  INCREASE_VAULT: 'increaseVault',
-  BALANCE: 'balance',
-  TICK: 'tick',
-  PRICE_FOR_INTERACTION: 'priceForInteraction',
-  EPOCH: 'epoch',
-};
+export type IOContractReadInteractions =
+  | 'gateway'
+  | 'gateways'
+  | 'prescribedObservers'
+  | 'record'
+  | 'auction'
+  | 'balance'
+  | 'epoch'
+  | 'priceForInteraction';
+
+export type IOContractWriteInteractions =
+  | 'delegateStake'
+  | 'decreaseDelegateStake'
+  | 'joinNetwork'
+  | 'leaveNetwork'
+  | 'increaseOperatorStake'
+  | 'decreaseOperatorStake'
+  | 'updateGatewaySettings'
+  | 'saveObservations'
+  | 'submitAuctionBid'
+  | 'buyRecord'
+  | 'extendRecord'
+  | 'increaseUndernameCount'
+  | 'transfer'
+  | 'vaultedTransfer'
+  | 'createVault'
+  | 'extendVault'
+  | 'increaseVault'
+  | 'tick';
+
+export type IOContractInteractions =
+  | IOContractReadInteractions
+  | IOContractWriteInteractions;
+
+export type IOContractInteractionsWithIOFees =
+  | 'buyRecord'
+  | 'submitAuctionBid'
+  | 'extendRecord'
+  | 'increaseUndernameCount';
```

Comment:
should use a `Pick` of the IOContractWriteInteractions

Reply:
We cant - pick consumes a record not a string type.

I mean, it CAN consume a "string type" as String, the class, but thats not what we want.